### PR TITLE
feat: update to ng18

### PR DESCRIPTION
**Pull request by Erbenos**
## PR Checklist

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[x] Other... Please describe: Update Dependencies
```

## Does this PR introduce a breaking change?

```
[X] Yes
[ ] No
```

Now depends on \~6 months old major version of Angular instead of \~12 months old one. No changes in code were necessary, all e2e tests still pass. Really only bumped dependencies.

Basically I want to update my own application thats using this library to Angular 18, but given this is a dependency of mine 
that up to this day depends on Angular ^17, it needs to be updated first. I noticed that #157 exists after I already did the change, and this unlike that one does not combine both unneeded behavior changes with dependency updates. Which should hopefully make it easier to approve given the #157 is yet to be merged nearly a quarter after it was opened.

### CODE REVIEW

**Comment by Erbenos** [comment]
I am moving it to draft cause I have noticed that the "Custom Template" part of playground does not correctly update its visibility state. This isn't caught by e2e tests.

File: projects/ngneat/helipopper/package.json [hunk 1]
```diff
@@ -4,11 +4,11 @@
   "description": "A Powerful Tooltip and Popover for Angular Applications",
   "dependencies": {
     "tslib": "^2.3.0",
-    "@ngneat/overview": "^6.0.0",
+    "@ngneat/overview": "6.1.1",
     "tippy.js": "6.3.7"
   },
   "peerDependencies": {
-    "@angular/core": ">=17.0.0"
```
Comment (NetanelBasal) [inline projects/ngneat/helipopper/package.json:11]: Any breaking changes that we need to change the min version?

**Comment by Erbenos** [comment]
Ughhh, I am stupid, nevermind. I was only looking at the root package.json not realizing the published package will get the nested one and the nested one specifies Angular >=17.0.0, you can in fact install it with Angular 18 just fine. My bad. Closing this.
